scripts/recovery/protect-and-retire-noncanonical-worktrees.sh

```
@@ -0,0 +1,392 @@
+  git -C "$repo" push -u origin "$fallback" --no-verify
+  BACKUP_REF["$repo"]="$fallback"
+}
+
+protect_submodules() {
+  local repo="$1"
+  local label="$2"
+  local subpath
+  while IFS= read -r subpath; do
+    [[ -n "$subpath" ]] || continue
+    protect_repo "$repo/$subpath" "$label-$(safe_name "$subpath")"
+  done < <(git -C "$repo" submodule foreach --quiet --recursive 'printf "%s\n" "$sm_path"' 2>/dev/null || true)
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)